src/main/java/de/telran/blog/service/TagsService.java

```diff
@@ -0,0 +1,41 @@
+package de.telran.blog.service;
+
+import de.telran.blog.dto.TagsDto;
+import de.telran.blog.entity.TagsEntity;
+import de.telran.blog.repository.TagsRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class TagsService {
+
+    private TagsRepository tagsRepository;
+
+    public TagsService(TagsRepository tagsRepository){
+        this.tagsRepository = tagsRepository;
+    }
+
+    public Long createTag(TagsDto dto ) {
+        TagsEntity tagsEntity = new TagsEntity();
+        tagsEntity.setText(dto.getText());
+        tagsEntity.setDescription(dto.getDescription());
+
+        return tagsRepository.save(tagsEntity).getId();
+    }
+
+    public List<TagsDto> getAllTags() {
+        return tagsRepository.findAll()
+                .stream()
+                .map(TagsDto::new)
+                .collect(Collectors.toList());
+    }
+
+    public TagsDto getDto( Long tagId  ) {
+        return new TagsDto(tagsRepository.findById(tagId)
+                .orElseThrow(
+                        () -> new RuntimeException("Tag with id=" + tagId + "not found")
+                ));
```

Comment:
@venividiviciregina as Alexander suggested, better not to use Dto Objects on the Service side, because Service should only use Entity Objects and all conversions from Dto to Entity and vice versa should be implemented on the Controller side.